src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,93 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75;
+
+    private Node<K, V>[] table;
+    private int realCapacity;
+    private int threshold;
+    private int size;
+
+    MyHashMap() {
+        realCapacity = DEFAULT_CAPACITY;
+        threshold = (int) (realCapacity * DEFAULT_LOAD_FACTOR);
+        table = new Node[realCapacity];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size == threshold) {
+            resize();
+        }
+        int bucket = getBucketNumber(key);
+        if (table[bucket] == null) {
+            table[bucket] = new Node<>(key, value);
+            size++;
+            return;
+        }
+        Node<K, V> currentNode = getNodeFromCollision(table[bucket], key);
+        if (Objects.equals(currentNode.key, key)) {
+            currentNode.value = value;
+        } else {
+            currentNode.next = new Node<>(key, value);
+            size++;
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        int bucket = getBucketNumber(key);
+        if (table[bucket] == null) {
+            return null;
+        }
+        Node<K, V> currentNode = getNodeFromCollision(table[bucket], key);
+        return Objects.equals(currentNode.key, key) ? currentNode.value : null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private Node<K, V> getNodeFromCollision(Node<K, V> node, K key) {
+        while (node.next != null) {
+            if (Objects.equals(node.key, key)) {
```

Comment:
it is not very good that you check for equality twice - once here, and then once more after you call this method in `put()` and `get()`

let's try to remove this method maybe